presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageFilePageSource.java

```diff
@@ -45,13 +44,26 @@
 
     public PageFilePageSource(
             FSDataInputStream inputStream,
+            long start,
+            long splitLength,
+            long fileSize,
             PagesSerde pagesSerde,
             List<HiveColumnHandle> columns)
+            throws IOException
     {
         this.inputStream = requireNonNull(inputStream, "inputStream is null");
-        pageReader = readPages(
-                requireNonNull(pagesSerde, "pagesSerde is null"),
-                new InputStreamSliceInput(inputStream));
+        PageFileFooterReader pageFileFooterReader = new PageFileFooterReader(inputStream, fileSize);
+
+        long[] readStartAndLength = getReadStartAndLength(
```

Comment:
Let's have a class instead of using an array. You can make it internal and private. e.g.:

```
private static class OffsetAndLength {
private final long offset;
private final long length;
...
}
```